apps/dev-workbench/workbench.js

```diff
@@ -0,0 +1,224 @@
+  //   data: files,
+  //   dataType: 'zip',
+  //   processData: false,
+  //   success: function(response) {
+  //     console.log(response);
+  //   },
+  // });
+  for (var i = 0; i < files.length; i++) {
+    handleFile(files[i]);
+  }
+});
+function resetLabelsModel() {
+  $('#labelsSubmitButton').prop('disabled', false);
+  $('#labelsSubmitText').show();
+  $('#labelsSubmitLoading').hide();
+  $('.labelsInputGroup').show();
+  $('#labelsFilterList').html('');
+  $('#filterLabels').hide();
+  $('#labelsDatasetZip').hide();
+  $('#labelsSubmitFinish').hide();
+  $('#labelsUploadForm').unbind('submit');
+  $('#labelsUploadForm').trigger('reset');
+
+  $('#labelsUploadForm').on('submit', function() {
+    $('#labelsSubmitButton').prop('disabled', true);
+    $('#labelsSubmitText').hide(200);
+    $('#labelsSubmitLoading').show(200);
+    let fileNames = $.map($('#labelsInput').prop('files'), function(val) {
+      return val.name;
+    });
+    var data = new FormData();
+    var fileInput = document.getElementById('labelsInput');
+    for (i = 0; i < fileNames.length; i++) {
+      data.append('zips', fileInput.files[i], fileNames[i]);
+    }
+    let url = '';
+    $.ajax({
+      type: 'POST',
+      url: '../../workbench/deleteDataset',
+      success: function(done1) {
+        console.log(done1);
+        if (fileNames.length == 1) {
+          url = '../../workbench/getLabelsZip';
+        } else if (fileNames.length > 1) {
+          url = '../../workbench/getLabelsZips';
+        } else return;
+        $.ajax({
+          type: 'POST',
+          enctype: 'multipart/form-data',
+          url: url,
+          data: data,
+          processData: false,
+          contentType: false,
+          cache: false,
+          timeout: 600000,
+          success: function(data) {
+            console.log(data);
+            $('#labelsSubmitButton').prop('disabled', false);
+            $('#labelsSubmitText').show(200);
+            $('#labelsSubmitLoading').hide(200);
+            displayLabels(data);
+          },
+          error: function(e) {
+            console.log('ERROR : ', e);
+            alert('Error');
+            $('#labelsSubmitButton').prop('disabled', false);
+            $('#labelsSubmitText').show(200);
+            $('#labelsSubmitLoading').hide(200);
+          },
+        });
+      },
+    });
+  });
+}
+
+function displayLabels(data) {
+  $('.labelsInputGroup').hide(180);
+  $('#labelsFilterList').html('');
+  for (let i = 0; i < data.labels.length; i++) {
+    $('#labelsFilterList').append(
+        ` <li
+      class="list-group-item d-flex justify-content-between align-items-center" >
+      <div class="custom-control custom-checkbox">
+        <input
+          type="checkbox"
+          class="custom-control-input"
+          id="labelCheck` +
+        i +
+        `" checked /> <label class="custom-control-label" for="labelCheck` +
+        i +
+        `" >` +
+        data.labels[i] +
```

Comment:
Can you either sanitize or use innerText to prevent xss for things like this?

Reply:
Yes, I should do that. Thanks for pointing out.

Reply:
I sanitised the user inputs accordingly. I think that should be safer?

Reply:
I'm not sure which would be safer, but as far as I'm concerned, both are fine. :smile: